kratos/includes/shared_pointers.h

```diff
@@ -19,18 +19,30 @@
 
 /* External includes */
 #include <memory>
+#include "intrusive_ptr/intrusive_ptr.hpp"
 
 namespace Kratos {
 
 template<class T>
-using shared_ptr = std::shared_ptr<T>; //std::shared_ptr<T>;
+using intrusive_ptr = std::intrusive_ptr<T>; 
 
 template<class T>
-using weak_ptr = std::weak_ptr<T>; //std::weak_ptr<T>;
+using intrusive_weak_ptr = std::intrusive_weak_ptr<T>; 
```

Comment:
shouldn't you also add `Kratos::make_intrusive_weak`(or maybe `make_weak_intrusive`)?

and what abt the typedefs line 80 and following?

Reply:
This function (if it existed at all) makes no sense. As `std::make_shared()` and `std::make_unique()`, an object with dynamic storage duration is allocated. As the function will return a weak reference, no other strong reference could ever exist after the function returns; so the object is destroyed immediately and the function returns an empty `weak_ptr`. What is the use of it?

Reply:
thanks @lhmouse , I was not thinking abt this
ofc you are right, my suggestion does not make sense

@RiccardoRossi still the question abt the typedefs at the end of the file are valid I think

Reply:
@philbucher i intentionally avoided redefining the macro below. My idea was that one would have to redefine the pointer definitions on its own as i did here

https://github.com/KratosMultiphysics/Kratos/pull/3255/files/f002f1b8e0a18e106dfbdf7518d216f7bcd15666#diff-f4a878a373c9d86ca79967299620406cR93

the rationale is that it is not enough to define the pointer, you also need to derive the class from intrusive_ptr_base.

My point is that i would prefer to minimize macros if i can. And i think we should use the intrusive_ptr only in very selected classes. Having said this, if there is consensus that defining 
 
        KRATOS_CLASS_INTRUSIVE_POINTER_DEFINITION

is a better option i will add it...